src/commands/command-lib/mod/streammessage.cmdlib.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2024 Inimi | DrHooBs | Incoverse
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import * as Discord from "discord.js";
+import { DrBotGlobal } from "@src/interfaces/global.js";
+import moment from "moment-timezone";
+import storage from "@src/lib/utilities/storage.js";
+import { DrBotSlashCommand } from "@src/lib/base/DrBotCommand.js";
+import { DrBotSubcommand } from "@src/lib/base/DrBotSubcommand.js";
+
+declare const global: DrBotGlobal;
+
+export default class ModStreamMessage extends DrBotSubcommand {
+    static parentCommand: string = "Mod";
+
+    public async setup(
+        parentSlashCommand: Discord.SlashCommandBuilder
+    ): Promise<boolean> {
+        parentSlashCommand.addSubcommand((subcommand) =>
+            subcommand
+                .setName("streammessage")
+                .setDescription("Set a stream message for a role")
+                .addStringOption((option) =>
+                    option
+                        .setName("message")
+                        .setDescription("Stream Message")
+                        .setRequired(true)
+                )
+                .addStringOption((option) =>
+                    option
+                        .setName("time")
+                        .setDescription("Time for the stream message")
+                        .setRequired(true)
+                )
+                .addRoleOption((option) =>
+                    option.setName("role").setDescription("Select a role")
+                )
+                .addStringOption((option) =>
+                    option
+                        .setName("template")
+                        .setDescription("Select a template")
+                        .addChoices({
+                            name: "stream", value: "stream"
+                        }, {
+                            name:"cancellation", value: "cancellation"})
+                )
+        );
+        this._loaded = true;
+        return true;
```

Comment:
A better solution here, which I myself haven't adopted too well but I think its better to start, is to do the following instead of the ``this._loaded = true`` and the ``return true``:

```suggestion
        return super.setup();
```
`super#setup` basically does ``this._loaded = true`` and ``return true``, so it makes it a bit better to read